src/app.ts

```diff
@@ -1,26 +1,54 @@
 // example interfaces that can be use
 // TIP: the types mentioned in the interfaces must be fulfilled in order to solve the problem.
-interface TemperatureReading {
-  time: Date
-  temperature: number
-  city: string
-}
-interface TemperatureSummary {
-  first: number
-  last: number
-  high: number
-  low: number
-  average: number
-}
+import { TemperatureReading, TemperatureSummary } from './types'
+
+const temperatureSummaries: Record<string, Record<number, TemperatureSummary>> =
+  {}
 
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  readings.forEach(({ city, temperature, time }) => {
+    const timeKey = time.getTime()
+
+    // Here I decided to use the nullish coalescing assignment (??=)
+    // because it assigns a default value ({}) to a property (temperatureSummaries[city])
+    // only when that property is null or undefined.
+
+    const cityData = (temperatureSummaries[city] ??= {})
+
+    const summary = (cityData[timeKey] ??= {
+      first: temperature,
+      last: temperature,
+      high: temperature,
+      low: temperature,
+      average: 0,
+      count: 0,
+    })
+
+    const { high, low } = summary
+
+    summary.last = temperature
+    summary.high = Math.max(temperature, high)
+    summary.low = Math.min(temperature, low)
+    summary.average += temperature
```

Comment:
Being very strict with the variable's names, this is not the average; it is the sum of all the temperatures, so I recommend using a more descriptive name to avoid confusion.